components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/file/FilePollingConsumer.java

```diff
@@ -723,36 +721,43 @@ private boolean acquireLock(FileSystemManager fsManager, FileObject fileObject)
      * @throws AxisFault
      */
     private FileObject processFile(FileObject file) throws AxisFault {
-        try {
-            FileContent content = file.getContent();
-            String fileName = file.getName().getBaseName();
-            String filePath = file.getName().getPath();
-            String fileURI = file.getName().getURI();
+        boolean isFailedRecord = VFSUtils.isFailRecord(fsManager, file);
```

Comment:
isFailedRecord() Why are you checking this again? Already done on top. This is an additional io operation. Also the formatting is not done properly.